### DISCUSSION TITLE
Niri Session Startup Script Issue

### DISCUSSION THREAD

**Discussion by WantenMN** [YaLTeR/niri#775] [Q&A] [answered]
(niri unstable 2024-10-30 (commit d3e72245b003cbeed0810bf00c4a445b7c8a652a))

I wrote a script in `~/.config/niri/test.sh` with the following content:

```bash
#!/usr/bin/env bash
echo $(date +"%H:%M:%S") > ~/Downloads/time.txt
```

I added this line at the end of the Niri `config.kdl` file:

```kdl
spawn-at-startup "bash ~/.config/niri/test.sh"
```
exited Niri, and logged back into my TTY session. However, when I launch Niri using `niri-session`, the `spawn-at-startup` setting has no effect, and no `time.txt` file is created.

**Comment by YaLTeR** [comment, marked as answer]
niri does not split arguments and does not expand ~ in arguments, you need to pass them separately: `spawn-at-startup "bash" "/home/USERNAME/.config/niri/test.sh"`

**Reply by sodiboo** [reply]
since the file in question has a shebang, it should also be sufficient to omit the interpreter from the invocation. `spawn-at-startup "~/.config/niri/test.sh"`. niri does expand `~` in the executable, which is exactly what the user wanted here :3

(if this doesn't work, make sure it's also actually executable: `chmod +x ~/.config/niri/test.sh`)
